include/crocoddyl/core/actuation/squashing/smooth-sat.hpp

```diff
@@ -0,0 +1,87 @@
+///////////////////////////////////////////////////////////////////////////////
+// BSD 3-Clause License
+//
+// Copyright (C) 2018-2020, University of Edinburgh, IRI: CSIC-UPC
+// Copyright note valid unless otherwise stated in individual files.
+// All rights reserved.
+///////////////////////////////////////////////////////////////////////////////
+
+#ifndef CROCODDYL_CORE_SQUASHING_SMOOTH_SAT_HPP_
+#define CROCODDYL_CORE_SQUASHING_SMOOTH_SAT_HPP_
+
+#include "crocoddyl/core/actuation/squashing-base.hpp"
+
+namespace crocoddyl {
+
+template <typename _Scalar>
+class SquashingModelSmoothSatTpl : public SquashingModelAbstractTpl<_Scalar> {
+ public:
+  typedef _Scalar Scalar;
+  typedef MathBaseTpl<Scalar> MathBase;
+  typedef SquashingModelAbstractTpl<Scalar> Base;
+  typedef SquashingDataAbstractTpl<Scalar> SquashingDataAbstract;
+  typedef typename MathBase::VectorXs VectorXs;
+
+  SquashingModelSmoothSatTpl(const Eigen::Ref<const VectorXs>& u_lb, const Eigen::Ref<const VectorXs>& u_ub,
+                             const std::size_t& ns)
+      : Base(ns) {
+    u_lb_ = u_lb;
+    u_ub_ = u_ub;
+
+    s_lb_ = u_lb_;
+    s_ub_ = u_ub_;
+
+    smooth_ = 0.1;
+
+    d_ = (u_ub_ - u_lb_) * smooth_;
+    a_ = d_.array() * d_.array();
+  }
+
+  ~SquashingModelSmoothSatTpl();
+
+  void calc(const boost::shared_ptr<SquashingDataAbstract>& data, const Eigen::Ref<const VectorXs>& s) {
```

Comment:
I think you removed the `virtual` keyword, it is a good practice to add it as well in derived classes.
```cpp
virtual void calc(const boost::shared_ptr<SquashingDataAbstract>& data, const Eigen::Ref<const VectorXs>& s) {
```